CMakeLists.txt

```diff
@@ -273,7 +273,10 @@ else()
   message(STATUS "Building libbinaryen as shared library.")
   add_library(binaryen SHARED ${binaryen_SOURCES} ${binaryen_objs})
 endif()
-install(TARGETS binaryen DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(TARGETS binaryen
+  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
```

Comment:
as this is a library, doesn't only the middle one here matter? I might be missing something...

Reply:
On the Windows platform, shared libraries have a runtime component (* .dll file), which is usually located in the 'bin' directory (near executable programs).

Reply:
Hmm.. but that isn't true on non-windows platforms.   `.so` files should not live in the `bin` directory on other platforms.

I find it odd these this stuff isn't all just handled by default with cmake.

Reply:
The other parts of this change LGTM for sure so you might want split of this part ?